### PR TITLE
Refactor Social Bundle

### PR DESCRIPTION
**Pull request by anton-vlasenko**
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/8013
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR removes deprecated code from Social Bundle to make it compatible with Symfony 3.x

### CODE REVIEW

File: plugins/MauticSocialBundle/Form/Type/FacebookType.php [hunk 1]
```diff
@@ -25,7 +26,7 @@ class FacebookType extends AbstractType
      */
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
-        $builder->add('layout', 'choice', [
+        $builder->add('layout', ChoiceType::class, [
             'choices' => [
                 'standard'     => 'mautic.integration.Facebook.share.layout.standard',
```
Comment (escopecz) [inline plugins/MauticSocialBundle/Form/Type/FacebookType.php]: Switch the labels and values for all choice types in this bundle as well. Use the `choices_as_values` option to simulate Symfony 3. See https://docs.google.com/document/d/1lHNK8hHQtvyWlz9OtfAoLupDVb5-QunQjq-HAGlAsS4/edit# for more details.

Comment (anton-vlasenko) [reply plugins/MauticSocialBundle/Form/Type/FacebookType.php]: Fixed d9df62781e4c91fa9191c9b2896ff018de22fed2

**Comment by escopecz** [comment]
@anton-vlasenko do you have a reaction to Ruth's comment? I can replicate.

The SocialIntegration loads the soclial config forms dynamically. Test for example by opening the Twitter plugin configuration.

The SocialIntegration class must instead define a new abstract method for example called `getForm()` that would return the form type class name. Then you'll define the form type for all classes that extend the SocialIntegration and then you can get rid of the dynamic loading from Factory and use the new `getForm` method instead.

**Comment by anton-vlasenko** [comment]
> @anton-vlasenko do you have a reaction to Ruth's comment? I can replicate.
> 
> The SocialIntegration loads the soclial config forms dynamically. Test for example by opening the Twitter plugin configuration.
> 
> The SocialIntegration class must instead define a new abstract method for example called `getForm()` that would return the form type class name. Then you'll define the form type for all classes that extend the SocialIntegration and then you can get rid of the dynamic loading from Factory and use the new `getForm` method instead.

> @anton-vlasenko do you have a reaction to Ruth's comment? I can replicate.

I'm working on this right now.

> The SocialIntegration class must instead define a new abstract method for example called `getForm()` that would return the form type class name.

Yes, I know.
That's exactly how I refactored another module.
I'm working on it. I will ask you for help if I don't resolve this issue in the reasonable amount of time.

**Comment by anton-vlasenko** [comment]
@RCheesley Thank you for testing this PR. The issue your mentioned should be fixed now.
 https://github.com/mautic/mautic/pull/8085/commits/acf13809f44fe959699984f60667bfb525f87fa9

**Comment by RCheesley** [comment]
> @RCheesley Thank you for testing this PR. The issue your mentioned should be fixed now.
> [acf1380](https://github.com/mautic/mautic/commit/acf13809f44fe959699984f60667bfb525f87fa9)

Awesome, will test it tomorrow!

**Pull request merged**